protos/sigstore_bundle.proto

```diff
@@ -63,7 +63,7 @@ message VerificationMaterial {
 message Bundle {
         // MUST be application/vnd.dev.sigstore.bundle+json;version=0.1
         // when encoded as JSON.
-        string media_type = 1;
+        string media_type = 1 [(google.api.field_behavior) = REQUIRED];
```

Comment:
Possibly irrelevant, but is anybody familiar with what this annotation implies about non-JSON-serialized protobufs? My understanding is that `media_type` is only sensible when encoded as JSON, so it may not make sense to set it as `REQUIRED` here.

Reply:
Let me try it out!

Reply:
So I was digging around a bit, and was not able to get a program to fail marshalling to either binary or JSON when the preconditions where not met (i.e some `REQUIRED` fields was not present). This led me to the [documentation](https://google.aip.dev/203) where I found this:

> **Note**: The vocabulary given in this document is for descriptive purposes only, and does not itself add any validation. The purpose is to consistently document this behavior for users.

So it any developer would still need to perform any validation manually it seems. The generated (Go) code does not seem to include any annotations for JSON marshal/unmarshal that would imply that a given field is `REQUIRED` or not.

Maybe this is old news to everyone, but it was not for me (my previous example is to rely on `optional` keywords and treat anything else as `REQUIRED`).

But to go back to @woodruffw 's question. I think it still make sense to include the `media_type` even when binary marshalling is being conducted, as we may not know what a downstream consumer is intending to do with the message. Maybe there is a service pulling the data from a central registry and serializing to disk (via JSON) for further consumption by a different service?

Reply:
@haydentherapper you got any different experiences with `REQUIRED` fields?

Reply:
This is correct, we need to find a library for enforcing this (I know of the internal google one, not sure what’s available for OSS)

Reply:
Java client does not verify `REQUIRED` fields, however, there's API to get the annotations: https://github.com/sigstore/sigstore-java/pull/317/files#diff-4a847f4c6f9e37621c0d9d5da4a86c027037a2c63052205b6471a662724e5b73R130-R136

Reply:
https://github.com/mwitkow/go-proto-validators - this looks a lot like the internal library

Reply:
I used protoc-gen-validate in an old life.

Reply:
What if it were a fixed string? Or at least an enum? That maybe prevents clients or clarifies usage in code that it must be certain values. 

At the same time, I do want it as a type hint. So yeah, maybe not required, but otherwise I don't know how to package this unless it's inside another wrapped proto with (payload=bundle, media_type)

Reply:
Agreed -- IMO it's not ideal, but the alternatives (hooking into the marshalling, or doing it after marshalling) are also probably more error prone/more likely to be forgotten due to a lack of controls.

I'm loathe to add *yet another* layer of schematization here, but maybe what we really need is a JSON Schema definition that gets generated from these protobuf definitions. That schema could then include (and enforce) the requirement hints. 

Just from a brief look, it seems like https://github.com/chrusty/protoc-gen-jsonschema might do the trick, and has support for their own `required` annotation to mark fields as non-optional.

Reply:
Works for me. This is definitely not something that should block any other work here, and I can pursue it as an independent follow-up.